src/main/webapp/queue.js

```diff
@@ -0,0 +1,182 @@
+    .collection('information')
+    .doc('queue').collection('videos')
+    .onSnapshot(function(snapshot) {
+      snapshot.docChanges().forEach(function(change) {
+        if (change.type === 'added') {
+          console.log('added: ', change.doc.data());
+        }
+        if (change.type === 'modified') {
+          console.log('modified: ', change.doc.data());
+        }
+        if (change.type === 'removed') {
+          console.log('removed: ', change.doc.data());
+        }
+      });
+      getRoomQueue('47jGbulshBCjcc8YOt8a');
+    });
+
+/* exported verifyURLStructure */
+/*
+  Makes sure the stub of the user's URL is valid and will warn user
+  if their link wasn't formatted correctly for the server to parse.
+*/
+function verifyURLStructure(url) {
+  const validator = new RegExp('' +
+    /(^(?:https?:\/\/)?(?:www\.)?)/.source +
+    /((?:youtu\.be\/|youtube\.com\/))/.source +
+    /((?:embed\/|v\/|watch\?v=|watch\?.+&v=))/.source +
+    /((\w|-){11})(?:\S+)?$/.source,
+  ); // regex for youtube link validation
+  if (!url.match(validator)) { // signal bad link to user
+    document.getElementById('linkError').style.display = 'block';
+    document.getElementById('videoError').style.display = 'none';
+  } else if (document.getElementById('ytplayer') != null) {
+    getVideoData(document.getElementById('linkArea').value.substring(32));
+  }
+}
+
+// Retrieve video and create a new room via servlet GET request
+async function getVideoData(id) {
+  if (id == '') {
+    console.log('NO ID');
+    return;
+  }
+  fetch('/vSearch?id=' + id)
+      .then((response) => response.json())
+      .then((video) => {
+        if (video.error == null) { // video was found, add to firestore
+          console.log('VIDEO FOUND');
+          document.getElementById('linkError').style.display = 'none';
+          document.getElementById('videoError').style.display = 'none';
+        } else if (video.items.length == 0) { // video not found, error
+          console.log('NO VIDEO FOUND');
+          document.getElementById('linkError').style.display = 'none';
+          document.getElementById('videoError').style.display = 'block';
+        }
+        console.log(video);
+      });
+  document.getElementById('linkArea').value = '';
+}
+
+/* exported removeVideo */
+/**
+ * Determines if the given video name exists within the database, then deletes
+ * it if found.
+ */
+async function removeVideo(roomid, name) {
+  const selectedVideo = await db // eslint-disable-line no-undef
+      .collection('rooms')
+      .doc(roomid)
+      .collection('information')
+      .doc('queue')
+      .collection('videos')
+      .doc(name)
+      .get();
+  if (selectedVideo.exists) {
+    await db // eslint-disable-line no-undef
+        .collection('rooms')
+        .doc(roomid)
+        .collection('information')
+        .doc('queue')
+        .collection('videos')
+        .doc(name)
+        .delete();
+    console.log('Video ' + name + ' deleted');
+  }
```

Comment:
If a user were to add two of the same video would deleting one of them cause both of them to be deleted? It looks like your deleting videos by name so I wasn't sure.